R/estimate_slopes.R

```diff
@@ -106,9 +119,23 @@ estimate_slopes.merMod <- estimate_slopes.lm
 
   # Basis
   # Sometimes (when exactly?) fails when transform argument is passed
-  trends <- tryCatch(emmeans::emtrends(model, levels, var = trend, transform = transform, ...),
-    error = function(e) emmeans::emtrends(model, levels, var = trend, ...)
-  )
+  trends <-
+    tryCatch(
+      {
+        if (component != "conditional") {
+          emmeans::emtrends(model, levels, var = trend, transform = transform, component = "zi", ...)
+        } else {
+          emmeans::emtrends(model, levels, var = trend, transform = transform, ...)
```

Comment:
@DominiqueMakowski I don't know why calling `emmeans::emtrends()` directly gives different results as compared to calling it from within `estimate_slopes()`... Both should be the same output:

``` r
require("modelbased")
#> Loading required package: modelbased
require("glmmTMB")
#> Loading required package: glmmTMB
require("emmeans")
#> Loading required package: emmeans

data(Salamanders)
model <- glmmTMB(
  count ~ cover + mined + (1 | site),
  ziformula = ~ cover + mined,
  family = truncated_poisson,
  data = Salamanders
)

estimate_slopes(model, trend = "cover")
#> mined | Coefficient |   SE |  df |         95% CI | Coefficient (std.)
#> ----------------------------------------------------------------------
#> yes   |       -0.20 | 0.06 | 637 | [-0.31, -0.09] |              -0.19
#> no    |       -0.20 | 0.06 | 637 | [-0.31, -0.09] |              -0.19

as.data.frame(emmeans::emtrends(model, "mined", var = "cover", transform = "response"))
#>   mined cover.trend         SE  df  lower.CL   upper.CL
#> 1   yes  -0.2463855 0.07305896 637 -0.389851 -0.1029200
#> 2    no  -0.7646025 0.24201824 637 -1.239852 -0.2893524
```

<sup>Created on 2020-06-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
@mattansb any idea?

Reply:
Sorry haven't had time to look at it, will do!

Reply:
Without `transform = "response"` they both have the same results:

```R
as.data.frame(emmeans::emtrends(model, "mined", var = "cover"))
#>   mined cover.trend         SE  df   lower.CL    upper.CL
#> 1   yes  -0.1992178 0.05743355 637 -0.3119998 -0.08643586
#> 2    no  -0.1992178 0.05743355 637 -0.3119998 -0.08643586
```

Reply:
> Without transform = "response" they both have the same results:

strange thing is, though, that `transform = "response"` by default in `estimate_slopes()`.

Reply:
sooo it means everything's okay?

Reply:
yes

Reply:
required really an ugly hack (https://github.com/easystats/modelbased/pull/67/commits/bd2c701d228f19f92cb13f4242bd47b04a3e85ae), but.... it seems to work!